wheels macos x86_64: gmp: could not find a working compiler

Build gmp in side A is OK, but in side B:

```
  [gmp-6.3.0]   [spkg-install] ./spkg-install: line 144: ./configfsf.guess: Permission denied
  [gmp-6.3.0]   [spkg-install] Checking what CFLAGS GMP would use if they were empty...
  [gmp-6.3.0]   [spkg-install] checking build system type... ivybridge-apple-darwin24.6.0
  [gmp-6.3.0]   [spkg-install] checking host system type... ivybridge-apple-darwin24.6.0
  [gmp-6.3.0]   [spkg-install] checking for a BSD-compatible install... /usr/bin/install -c
  [gmp-6.3.0]   [spkg-install] checking whether build environment is sane... yes
  [gmp-6.3.0]   [spkg-install] checking for a thread-safe mkdir -p... ./install-sh -c -d
  [gmp-6.3.0]   [spkg-install] checking for gawk... no
  [gmp-6.3.0]   [spkg-install] checking for mawk... no
  [gmp-6.3.0]   [spkg-install] checking for nawk... no
  [gmp-6.3.0]   [spkg-install] checking for awk... awk
  [gmp-6.3.0]   [spkg-install] checking whether make sets $(MAKE)... yes
  [gmp-6.3.0]   [spkg-install] checking whether make supports nested variables... yes
  [gmp-6.3.0]   [spkg-install] checking whether to enable maintainer-specific portions of Makefiles... no
  [gmp-6.3.0]   [spkg-install] checking ABI=64
  [gmp-6.3.0]   [spkg-install] checking compiler gcc -std=gnu23 -O2 -pedantic -fomit-frame-pointer -m64 ... no, mpn_lshift_com optimization
  [gmp-6.3.0]   [spkg-install] checking ABI=x32
  [gmp-6.3.0]   [spkg-install] checking compiler gcc -std=gnu23 -O2 -pedantic -fomit-frame-pointer -mx32 ... no, mpn_lshift_com optimization
  [gmp-6.3.0]   [spkg-install] checking ABI=32
  [gmp-6.3.0]   [spkg-install] checking compiler gcc -std=gnu23 -m32 -O2 -pedantic -fomit-frame-pointer ... no
  [gmp-6.3.0]   [spkg-install] checking compiler gcc -std=gnu23 -O2 -pedantic -fomit-frame-pointer ... no, mpn_lshift_com optimization
  [gmp-6.3.0]   [spkg-install] configure: error: could not find a working compiler, see config.log for details
  [gmp-6.3.0]   [spkg-install] Error configuring GMP (with CFLAGS unset).
```
https://github.com/passagemath/passagemath/actions/runs/23609961449/job/68800764304#step:15:14315

--- a/build/pkgs/gmp/spkg-install.in
+++ b/build/pkgs/gmp/spkg-install.in
@@ -23,6 +23,8 @@ case "$MSYSTEM" in
         ;;
 esac
 
+export CC="$CC -std=gnu17"
+
 user_cflags=$CFLAGS # Save them. 'sage-env' sets CC, but not CFLAGS.
 required_cflags=""  # Additional mandatory settings required by Sage, accumulated below.
 user_ldflags=$LDFLAGS # Save them.

